packages/location_background/example/lib/main.dart

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file
+
+import 'dart:isolate';
+import 'dart:ui' hide TextStyle;
+
+import 'package:flutter/material.dart';
+import 'package:location_background_plugin/location_background_plugin.dart';
+
+import 'background.dart';
+
+void main() {
+  runApp(new MyApp());
+}
+
+class MyApp extends StatefulWidget {
+  @override
+  _MyAppState createState() {
+    return new _MyAppState();
+  }
+}
+
+class _MyAppState extends State<MyApp> {
+  ReceivePort _foregroundPort = new ReceivePort();
+  LocationBackgroundPlugin _locationPlugin;
+  Location _lastLocation;
+  bool _isTracking = false;
+
+  @override
+  void initState() {
+    _lastLocation = new Location(-1.0, 0.0, 0.0, -1.0, -1.0);
+    super.initState();
+    initPlatformState();
+  }
+
+  void initPlatformState() {
+    // The IsolateNameServer allows for us to create a mapping between a String
+    // and a SendPort that is managed by the Flutter engine. A SendPort can
+    // then be looked up elsewhere, like a background callback, to establish
+    // communication channels between isolates that were not spawned by one
+    // another.
+    if (!IsolateNameServer.registerPortWithName(
+        _foregroundPort.sendPort, kLocationPluginPortName)) {
+      throw 'Unable to register port!';
+    }
+
+    // Listen on the port for location updates from our background callback.
+    _foregroundPort.listen((dynamic message) {
```

Comment:
When does this port get closed?

Reply:
This port *should* stay open the life of the UI, but I've updated this logic to remove the port from the `IsolateNameServer` on close.